CMakeLists.txt

```diff
@@ -1284,6 +1286,37 @@ else()
   list(APPEND Sources ${GUI_SOURCES} ${GUI_HEADER_ONLY} ${GUI_UIS} ${GUI_HEADERS})
 endif()
 
+if(ENABLE_GUI_TESTS)
+  set(GUITESTS_HEADERS
+    src/guitests/guitests.h
+    src/guitests/TestModuleCache.h
+    src/guitests/TestTabManager.h
+    src/guitests/TestMainWindow.h
+    src/guitests/UXTest.h
+  )
+
+  set(GUITESTS_SOURCES
+    src/guitests/guitests.cc
+    src/guitests/TestModuleCache.cc
+    src/guitests/TestTabManager.cc
+    src/guitests/TestMainWindow.cc
+    src/guitests/UXTest.cc
+  )
+
+  list(APPEND Sources ${GUITESTS_SOURCES} ${GUITESTS_HEADERS})
+  target_compile_definitions(OpenSCAD PRIVATE HAS_GUI_TESTS=true)
```

Comment:
For boolean flags like this, prefer just setting the macro; kill the `=true` part.

Reply:
Thank for the feedback,

For the record, I have seen the use in other codebase the presence/absence of the feature using constexpr like in
```cpp
if constexpr(HAS_GUI_TEST)
{
     do_something
}     
```
Which i found more readable and more robust as this part of the code is thus always syntatically compiled on CI (while macro are skipped). 

That's said... I will make the change to use the more classical #ifdef #endif

Reply:
Yeah, generally I agree that if we can find a way for compilers to be aware of the code that should be excluded, it would be better than leaving it to the preprocessor. It's a good topic to investigate as a pure cleanup task.